### ISSUE TITLE
steps_per_epoch not propagating to Keras callbacks in tf-nightly

### ISSUE THREAD

**Issue by tgaddair** [tensorflow/tensorflow#37237]
This behavior was tested in `tf-nightly==2.2.0.dev20200302` on macOS.

Some of our Keras callbacks rely on accessing `self.params['steps']` in order to take actions based on how far along the process is into the current epoch (e.g., learning rate schedule).  However, it appears that this param is no longer being set properly, even when `mode.fit(steps_per_epoch=...)` is called.

I did some digging, and it looks like the issue can be traced to the way `CallbackList(steps=...)` is being set [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/engine/training.py#L753).  The `DataAdapter` object is an instance of `DatasetAdapter`, which is initialized with the `steps_per_epoch` param and assigns it to `self._user_steps`, but the `get_size()` method always returns `None` [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/engine/data_adapter.py#L704).

Is there a workaround for this?  Ideally, we'd like for callbacks to have access to the `steps_per_epoch` information without having to pass it in to each callback manually.  

You can see an example of how we use this in Horovod in the LearningRateWarmupCallback from the [tensorflow2_keras_mnist.py](https://github.com/horovod/horovod/blob/master/examples/tensorflow2_keras_mnist.py#L77) example.  As you can see in the implementation of that callback [here](https://github.com/horovod/horovod/blob/cff4de331edcd6cabbd807b7010e7432b03fe2b7/horovod/_keras/callbacks.py#L108), we are attempting to access `self.params.get('step')`, which used to get set correctly prior to v2.2.

cc @reedwm @pkanwar23

**Comment by omalleyt12** [comment]
Thanks for the issue! Working on a fix now

**Comment by tgaddair** [comment]
Hey @omalleyt12, any update on this issue?

**Comment by omalleyt12** [comment]
@tgaddair Yes this should be fixed in the latest nightly

Also have a cherrypick out for the 2.2 release branch: https://github.com/tensorflow/tensorflow/pull/37314

**Comment by tgaddair** [comment]
Thanks!  I'll try it out today and get back to you.

**Comment by omalleyt12** [comment]
Sounds good, thanks!

**Comment by tgaddair** [comment]
Looks like this issue is resolved.  Thanks for the fix @omalleyt12!

**Comment by omalleyt12** [comment]
Great, no prob, thanks for verifying!
